sympy/geometry/tests/test_line.py

```diff
@@ -192,7 +192,7 @@ def test_line_geom():
     assert r4.intersection(r7) == r7.intersection(r4) == [r7]
 
     # Segment contains
-    a, b = symbols('a,b')
+    a, b = symbols('a,b', real=True)
     s = Segment((0, a), (0, b))
     assert Point(0, (a + b)/2) in s
```

Comment:
Good thinking.  I've replaced my containment check with one that relies on the triangle inequality instead of the dot product.